File: day_1/day_1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	// Part 1
	increases := 0
	depths := getInput("./day_1/day_1_input.txt")
	for i := 1; i <= len(depths)-1; i++ {
		if depths[i] > depths[i-1] {
			increases += 1
		}
	}
	fmt.Printf("Part 1: %v", increases)

	// Part 2
	partTwoIncreases := 0
	partTwoDepths := make([]int64, 0)
	var quickResult int64
	for i := 0; i <= len(depths)-3; i++ {
		quickResult = depths[i] + depths[i+1] + depths[i+2]
		partTwoDepths = append(partTwoDepths, quickResult)
	}

	for j := 1; j <= len(partTwoDepths)-1; j++ {
		if partTwoDepths[j] > partTwoDepths[j-1] {
			partTwoIncreases += 1
		}
	}
	fmt.Printf("Part 2: %v", partTwoIncreases)

}

func getInput(filename string) []int64 {
	lines := make([]int64, 0)

	f, err := os.Open(filename)
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	scanner := bufio.NewScanner(f)
	scanner.Split(bufio.ScanLines)

	for scanner.Scan() {
		intLine, err2 := strconv.ParseInt(scanner.Text(), 10, 64)
		if err2 != nil {
			log.Fatal(err2)
		}

		lines = append(lines, intLine)
	}
	return lines
}

File: day_1/day_1.py
#!/usr/bin/python3

def get_input():
    with open("./day_1/day_1_input.txt", "r") as f:
        return [int(line.strip()) for line in f]

def part_1(depth):
    return sum([1 for i in range(len(depth)) if depth[i] > depth[i-1]])

def part_2(depth):
    return [(depth[i] + depth[i+1] + depth[i+2]) for i in range(len(depth) - 2)]

if __name__ == "__main__":
    print(f'The solution to part 1 is {part_1(get_input())}. The solution to part 2 is {part_1(part_2(get_input()))}.')